oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/TraceQueryService.java

```diff
@@ -374,6 +390,14 @@ public Trace queryTrace(final String traceId) throws IOException {
                 rootSpans.add(span);
             }
         });
+        /**
+         * In some cases, there are segment fragments, which could not be linked by Ref,
+         * because of two kinds of reasons.
+         * 1. Multiple leaf segments have no particular order in the storage.
+         * 2. Lost in sampling, agent fail safe, segment lost, even bug.
+         * Sorting the segments makes the trace view more readable.
+         */
+        rootSpans.sort(Comparator.comparing(span -> new Long(span.getStartTime())));
```

Comment:
Seems no need to box `span.getStartTime()`, and can be simplified with method reference

Reply:
Fixed.